## Final Project Submission

Please fill out:
* Student name: KIGO VALERIE WAMWITHA
* Student pace: Full Time(HYBRID)
* Scheduled project review date/time: 05/07/2025
* Instructor name: FAITH ROTICH
* Blog post URL:


In [ ]:
# Your code here - remember to use markdown cells for comments as well!

## New_Movie_Sudio_Project - Final Project
The company views Original Video Content as something they can join and venture in. As a data scientist, the prospect is to translate the findings of which movies they should put in their New Movie Studio. Also, find actionable insights that they can use and give recommendations.
### Objectives
* To find the type of movies the company should create.
* To yield 3 concrete business recommendations that the business should take.
* To describe the project goals,data,methods and results to shareholders.

  
Will be able to deal with;

* Missing values in data
* How to aggregate data
* How to visualize and make data easily accessible

Use f 1025 Rws and 17 Columns for the analysis, which included variables about:
* Revenue - domestic_gross, foreign_gross,worldwide_gross and production_budget
* Data - Primary_title, studio,release_date,genres
* performance_measure- averagerating, numvotes
* Personnel - primary_name,primary_profession

#### Task: Analyze the type of films the company should create.

## Business Understanding
### What is the relationship between potential movie types and audience reviews or movie types that did well?
### What type of movie will meet the needs of the company?
Intuitively it may be assumed that a mvies with low reviews or low rates is considered not reliable.

In this analysis we are going to look into IMDB and Bom movies dataset that will help in determining which movie type will be the best to use.

In [1]:
import numpy as np
import pandas as pd 
import sqlite3
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns

# Data Understanding
The data source for this analysis will be from;
### 'AviationData.csv'
* *Sources* : The data was retrieved from Kaggle
*  *Contents* : Data about the potential risk of each flight. Contains; model numbers, fatal injuries and the accidents occured.
*  *Format* : It is a csv data
## TASK
* Check missing values
* Aggregate Data
## Use of
* df.unique(): to check if there are unique values and also replacing the unique values using nan.
* df.head(): show first rows of a dataframe.
* df.shape: to know the size of the data set.
* df.isnull(): to check for null values.
* df.dropna(): to drop subsets of null values.


In [2]:
df_bom = pd.read_csv('bom.movie_gross.csv')
df_bom.head(10)

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   
5                   The Twilight Saga: Eclipse   Sum.     300500000.0   
6                                   Iron Man 2   Par.     312400000.0   
7                                      Tangled     BV     200800000.0   
8                                Despicable Me   Uni.     251500000.0   
9                     How to Train Your Dragon   P/DW     217600000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010  
5     398000000  2010  
6     311500000  2010  
7     391000000  2010  
8     291600000  2010  
9     277300000  2010

In [3]:
df_bom.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [4]:
df_bom.describe().tail(10)

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

In [5]:
df_bom.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [6]:
df_bom[df_bom['foreign_gross'].isnull()].head()

title studio  domestic_gross  \
222                                 Flipped     WB       1800000.0   
254  The Polar Express (IMAX re-issue 2010)     WB        673000.0   
267                          Tiny Furniture    IFC        392000.0   
269           Grease (Sing-a-Long re-issue)   Par.        366000.0   
280                         Last Train Home  Zeit.        288000.0   

    foreign_gross  year  
222           NaN  2010  
254           NaN  2010  
267           NaN  2010  
269           NaN  2010  
280           NaN  2010

In [7]:
df_bom['foreign_gross'] = df_bom['foreign_gross'].fillna(0)

In [8]:
df_bom['domestic_gross'] = df_bom['domestic_gross'].fillna(0)

In [9]:

df_bom['studio'] = df_bom['studio'].fillna(0)

In [10]:
df_bom.isnull().sum()

title             0
studio            0
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

In [11]:
df_bom.duplicated().sum()

0

In [12]:
df_bom.nunique()

title             3386
studio             258
domestic_gross    1798
foreign_gross     1205
year                 9
dtype: int64

In [13]:
df_bom.shape

(3387, 5)

#### From the data set above, it is clear that the bom_movie_gross dataset has null values in domestic and foreign gross, also, in studio. This is cleaned by filling (0) on the missing values. The data set has a total of 3,387rows and 5 columns. 

## Cleaning The Numbers Movie Budgets Data

In [14]:
df_budget= pd.read_csv('tn.movie_budgets.csv')
df_budget.head(10)


id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
5   6  Dec 18, 2015         Star Wars Ep. VII: The Force Awakens   
6   7  Apr 27, 2018                       Avengers: Infinity War   
7   8  May 24, 2007   Pirates of the Caribbean: At World’s End   
8   9  Nov 17, 2017                               Justice League   
9  10   Nov 6, 2015                                      Spectre   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747  
5      $306,000,000   $936,662,225  $2,053,311,220  
6      $300,000,000   $678,815,482  $2,048,134,200  
7      $300,000,000   $309,420,425    $963,420,425  
8      $300,000,000   $229,024,295    $655,945,209  
9      $300,000,000   $200,074,175    $879,620,923

In [15]:
df_budget.tail(10)

id  release_date                          movie production_budget  \
5772  73  Jan 13, 2012                      Newlyweds            $9,000   
5773  74  Feb 26, 1993                    El Mariachi            $7,000   
5774  75   Oct 8, 2004                         Primer            $7,000   
5775  76  May 26, 2006                         Cavite            $7,000   
5776  77  Dec 31, 2004                The Mongol King            $7,000   
5777  78  Dec 31, 2018                         Red 11            $7,000   
5778  79   Apr 2, 1999                      Following            $6,000   
5779  80  Jul 13, 2005  Return to the Land of Wonders            $5,000   
5780  81  Sep 29, 2015           A Plague So Pleasant            $1,400   
5781  82   Aug 5, 2005              My Date With Drew            $1,100   

     domestic_gross worldwide_gross  
5772         $4,584          $4,584  
5773     $2,040,920      $2,041,928  
5774       $424,760        $841,926  
5775        $70,071         $71,644  
5776           $900            $900  
5777             $0              $0  
5778        $48,482        $240,495  
5779         $1,338          $1,338  
5780             $0              $0  
5781       $181,041        $181,041

In [16]:
df_budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [17]:
df_budget.describe()

id
count  5782.000000
mean     50.372363
std      28.821076
min       1.000000
25%      25.000000
50%      50.000000
75%      75.000000
max     100.000000

In [18]:
df_budget.dtypes

id                    int64
release_date         object
movie                object
production_budget    object
domestic_gross       object
worldwide_gross      object
dtype: object

In [19]:
df_budget.isnull().sum()

id                   0
release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

In [20]:
df_budget.duplicated().sum()

0

In [21]:
df_budget.nunique()

id                    100
release_date         2418
movie                5698
production_budget     509
domestic_gross       5164
worldwide_gross      5356
dtype: int64

In [22]:
df_budget.shape

(5782, 6)

In [23]:
df_budget['release_date']=pd.to_datetime(df_budget['release_date'])

In [24]:
df_budget.dtypes

id                            int64
release_date         datetime64[ns]
movie                        object
production_budget            object
domestic_gross               object
worldwide_gross              object
dtype: object

From the above, it is clear that The Numbers Movie budget dataset has no null values, no duplicates, a total of 5,782 rows, and 6 columns. The columns for the 'Release date' should be in a Date Time Format.

## Merging the tw data sets

In [25]:
df_bom.rename(columns = {'title':'movie'}, inplace= True)
df_bom.head()

movie studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [26]:
merged = pd.merge(df_bom,df_budget, on= 'movie', how='inner')
merged

movie studio  domestic_gross_x foreign_gross  year  \
0                    Toy Story 3     BV       415000000.0     652000000  2010   
1                      Inception     WB       292600000.0     535700000  2010   
2            Shrek Forever After   P/DW       238700000.0     513900000  2010   
3     The Twilight Saga: Eclipse   Sum.       300500000.0     398000000  2010   
4                     Iron Man 2   Par.       312400000.0     311500000  2010   
...                          ...    ...               ...           ...   ...   
1242                       Gotti     VE         4300000.0             0  2018   
1243                 Ben is Back  RAtt.         3700000.0             0  2018   
1244  Bilal: A New Breed of Hero     VE          491000.0       1700000  2018   
1245                       Mandy    RLJ         1200000.0             0  2018   
1246                Lean on Pete    A24         1200000.0             0  2018   

       id release_date production_budget domestic_gross_y worldwide_gross  
0      47   2010-06-18      $200,000,000     $415,004,880  $1,068,879,522  
1      38   2010-07-16      $160,000,000     $292,576,195    $835,524,642  
2      27   2010-05-21      $165,000,000     $238,736,787    $756,244,673  
3      53   2010-06-30       $68,000,000     $300,531,751    $706,102,828  
4      15   2010-05-07      $170,000,000     $312,433,331    $621,156,389  
...   ...          ...               ...              ...             ...  
1242   64   2018-06-15       $10,000,000       $4,286,367      $6,089,100  
1243   95   2018-12-07       $13,000,000       $3,703,182      $9,633,111  
1244  100   2018-02-02       $30,000,000         $490,973        $648,599  
1245   71   2018-09-14        $6,000,000       $1,214,525      $1,427,656  
1246   13   2018-04-06        $8,000,000       $1,163,056      $2,455,027  

[1247 rows x 10 columns]

In [27]:
df_bom.rename(columns = {'year':'release_date'}, inplace= True)
df_bom.head()

movie studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  release_date  
0     652000000          2010  
1     691300000          2010  
2     664300000          2010  
3     535700000          2010  
4     513900000          2010

In [28]:
df_bom['release_date']=pd.to_numeric(df_bom['release_date'], errors='coerce')
df_budget['release_date']=pd.to_datetime(df_budget['release_date'], errors='coerce')
df_budget['release_date'] = df_budget['release_date'].dt.year
df_bom['release_date'] = df_bom['release_date'].astype('int64')
df_budget['release_date'] = df_budget['release_date'].astype('int64')
merged_1 = pd.merge(df_bom,df_budget, on= 'release_date', how='inner')
merged_1

movie_x studio  domestic_gross_x foreign_gross  \
0             Toy Story 3     BV       415000000.0     652000000   
1             Toy Story 3     BV       415000000.0     652000000   
2             Toy Story 3     BV       415000000.0     652000000   
3             Toy Story 3     BV       415000000.0     652000000   
4             Toy Story 3     BV       415000000.0     652000000   
...                   ...    ...               ...           ...   
814794  An Actor Prepares  Grav.            1700.0             0   
814795  An Actor Prepares  Grav.            1700.0             0   
814796  An Actor Prepares  Grav.            1700.0             0   
814797  An Actor Prepares  Grav.            1700.0             0   
814798  An Actor Prepares  Grav.            1700.0             0   

        release_date  id                movie_y production_budget  \
0               2010  15                Tangled      $260,000,000   
1               2010  39             Robin Hood      $210,000,000   
2               2010  47            Toy Story 3      $200,000,000   
3               2010  51    Alice in Wonderland      $200,000,000   
4               2010  58           Tron: Legacy      $200,000,000   
...              ...  ..                    ...               ...   
814794          2018  77            Blood Feast        $1,200,000   
814795          2018  29   Unfriended: Dark Web        $1,000,000   
814796          2018   1  That Way Madness Lies          $650,000   
814797          2018  67           Higher Power          $500,000   
814798          2018  78                 Red 11            $7,000   

       domestic_gross_y worldwide_gross  
0          $200,821,936    $586,477,240  
1          $105,487,148    $322,459,006  
2          $415,004,880  $1,068,879,522  
3          $334,191,110  $1,025,491,110  
4          $172,062,763    $400,062,763  
...                 ...             ...  
814794           $8,708          $8,708  
814795       $8,866,745     $16,434,588  
814796           $1,447          $1,447  
814797             $528            $528  
814798               $0              $0  

[814799 rows x 10 columns]

In [29]:
merged_2= pd.merge(df_bom,df_budget, on=['movie', 'release_date'], how='inner')
merged_2

movie studio  domestic_gross_x foreign_gross  \
0                    Toy Story 3     BV       415000000.0     652000000   
1                      Inception     WB       292600000.0     535700000   
2            Shrek Forever After   P/DW       238700000.0     513900000   
3     The Twilight Saga: Eclipse   Sum.       300500000.0     398000000   
4                     Iron Man 2   Par.       312400000.0     311500000   
...                          ...    ...               ...           ...   
1210                       Gotti     VE         4300000.0             0   
1211                 Ben is Back  RAtt.         3700000.0             0   
1212  Bilal: A New Breed of Hero     VE          491000.0       1700000   
1213                       Mandy    RLJ         1200000.0             0   
1214                Lean on Pete    A24         1200000.0             0   

      release_date   id production_budget domestic_gross_y worldwide_gross  
0             2010   47      $200,000,000     $415,004,880  $1,068,879,522  
1             2010   38      $160,000,000     $292,576,195    $835,524,642  
2             2010   27      $165,000,000     $238,736,787    $756,244,673  
3             2010   53       $68,000,000     $300,531,751    $706,102,828  
4             2010   15      $170,000,000     $312,433,331    $621,156,389  
...            ...  ...               ...              ...             ...  
1210          2018   64       $10,000,000       $4,286,367      $6,089,100  
1211          2018   95       $13,000,000       $3,703,182      $9,633,111  
1212          2018  100       $30,000,000         $490,973        $648,599  
1213          2018   71        $6,000,000       $1,214,525      $1,427,656  
1214          2018   13        $8,000,000       $1,163,056      $2,455,027  

[1215 rows x 9 columns]

## Im.db Data Cleaning

In [30]:
conn = sqlite3.connect("im.db")
cur = conn.cursor()

In [31]:
df_IMDB_table =pd.read_sql ("""
        SELECT name
        FROM Sqlite_master
        WHERE type = 'table';
""",conn)
df_IMDB_table

name
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers

In [32]:
df_basics =pd.read_sql ("""
         SELECT *
         FROM movie_basics;
""",conn)
df_basics

movie_id                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                  None  
146143              NaN           Documentary  

[146144 rows x 6 columns]

In [33]:
df_basics.dtypes

movie_id            object
primary_title       object
original_title      object
start_year           int64
runtime_minutes    float64
genres              object
dtype: object

In [34]:
df_basics.isnull().sum()

movie_id               0
primary_title          0
original_title        21
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64

In [35]:
df_basics.dropna(subset = ['runtime_minutes'], inplace= True)
df_basics.isnull().sum()

movie_id              0
primary_title         0
original_title        4
start_year            0
runtime_minutes       0
genres             2172
dtype: int64

In [36]:
df_basics['genres']= df_basics['genres'].fillna('None')
df_basics.isnull().sum()

movie_id           0
primary_title      0
original_title     4
start_year         0
runtime_minutes    0
genres             0
dtype: int64

In [37]:
df_basics['original_title']= df_basics['original_title'].fillna('None')
df_basics.isnull().sum()

movie_id           0
primary_title      0
original_title     0
start_year         0
runtime_minutes    0
genres             0
dtype: int64

In [38]:
df_directors= pd.read_sql ("""
         SELECT *
         FROM directors;
""",conn)
df_directors

movie_id   person_id
0       tt0285252   nm0899854
1       tt0462036   nm1940585
2       tt0835418   nm0151540
3       tt0835418   nm0151540
4       tt0878654   nm0089502
...           ...         ...
291169  tt8999974  nm10122357
291170  tt9001390   nm6711477
291171  tt9001494  nm10123242
291172  tt9001494  nm10123248
291173  tt9004986   nm4993825

[291174 rows x 2 columns]

In [39]:
df_directors.dtypes

movie_id     object
person_id    object
dtype: object

In [40]:
df_directors.isnull().sum()

movie_id     0
person_id    0
dtype: int64

In [41]:
df_directors.duplicated().sum()
print(f" Duplicate rows: {df_directors}")



 Duplicate rows:          movie_id   person_id
0       tt0285252   nm0899854
1       tt0462036   nm1940585
2       tt0835418   nm0151540
3       tt0835418   nm0151540
4       tt0878654   nm0089502
...           ...         ...
291169  tt8999974  nm10122357
291170  tt9001390   nm6711477
291171  tt9001494  nm10123242
291172  tt9001494  nm10123248
291173  tt9004986   nm4993825

[291174 rows x 2 columns]


In [42]:
df_directors.duplicated(subset=['movie_id', 'person_id'])

0         False
1         False
2         False
3          True
4         False
          ...  
291169    False
291170    False
291171    False
291172    False
291173    False
Length: 291174, dtype: bool

In [43]:
df_directors.drop_duplicates(subset=['movie_id', 'person_id'])
df_directors

movie_id   person_id
0       tt0285252   nm0899854
1       tt0462036   nm1940585
2       tt0835418   nm0151540
3       tt0835418   nm0151540
4       tt0878654   nm0089502
...           ...         ...
291169  tt8999974  nm10122357
291170  tt9001390   nm6711477
291171  tt9001494  nm10123242
291172  tt9001494  nm10123248
291173  tt9004986   nm4993825

[291174 rows x 2 columns]

In [44]:
df_ratings = pd.read_sql ("""
         SELECT *
         FROM movie_ratings;
""",conn)
df_ratings.isnull().sum()

movie_id         0
averagerating    0
numvotes         0
dtype: int64

In [45]:
df_ratings.duplicated().sum()
print(f" Duplicate rows: {df_ratings}")

 Duplicate rows:          movie_id  averagerating  numvotes
0      tt10356526            8.3        31
1      tt10384606            8.9       559
2       tt1042974            6.4        20
3       tt1043726            4.2     50352
4       tt1060240            6.5        21
...           ...            ...       ...
73851   tt9805820            8.1        25
73852   tt9844256            7.5        24
73853   tt9851050            4.7        14
73854   tt9886934            7.0         5
73855   tt9894098            6.3       128

[73856 rows x 3 columns]


In [46]:
df_persons = pd.read_sql ("""
         SELECT *
         FROM persons;
""",conn)
df_persons.isnull().sum()

person_id                  0
primary_name               0
birth_year            523912
death_year            599865
primary_profession     51340
dtype: int64

In [47]:
df_persons.drop(columns = ['birth_year', 'death_year'], inplace= True)
df_persons

person_id         primary_name  \
0       nm0061671    Mary Ellen Bauder   
1       nm0061865         Joseph Bauer   
2       nm0062070           Bruce Baum   
3       nm0062195         Axel Baumann   
4       nm0062798          Pete Baxter   
...           ...                  ...   
606643  nm9990381         Susan Grobes   
606644  nm9990690          Joo Yeon So   
606645  nm9991320       Madeline Smith   
606646  nm9991786  Michelle Modigliani   
606647  nm9993380       Pegasus Envoyé   

                                      primary_profession  
0              miscellaneous,production_manager,producer  
1             composer,music_department,sound_department  
2                             miscellaneous,actor,writer  
3       camera_department,cinematographer,art_department  
4       production_designer,art_department,set_decorator  
...                                                  ...  
606643                                           actress  
606644                                           actress  
606645                                           actress  
606646                                          producer  
606647                             director,actor,writer  

[606648 rows x 3 columns]

In [48]:
merged_temp= pd.merge(df_basics,df_directors, on=['movie_id'], how='inner')
merged_temp

movie_id                    primary_title       original_title  \
0       tt0063540                        Sunghursh            Sunghursh   
1       tt0063540                        Sunghursh            Sunghursh   
2       tt0063540                        Sunghursh            Sunghursh   
3       tt0063540                        Sunghursh            Sunghursh   
4       tt0066787  One Day Before the Rainy Season      Ashad Ka Ek Din   
...           ...                              ...                  ...   
238744  tt9916190                        Safeguard            Safeguard   
238745  tt9916538              Kuambil Lagi Hatiku  Kuambil Lagi Hatiku   
238746  tt9916538              Kuambil Lagi Hatiku  Kuambil Lagi Hatiku   
238747  tt9916538              Kuambil Lagi Hatiku  Kuambil Lagi Hatiku   
238748  tt9916730                           6 Gunn               6 Gunn   

        start_year  runtime_minutes              genres   person_id  
0             2013            175.0  Action,Crime,Drama   nm0712540  
1             2013            175.0  Action,Crime,Drama   nm0712540  
2             2013            175.0  Action,Crime,Drama   nm0712540  
3             2013            175.0  Action,Crime,Drama   nm0712540  
4             2019            114.0     Biography,Drama   nm0002411  
...            ...              ...                 ...         ...  
238744        2019             90.0      Drama,Thriller   nm7308376  
238745        2019            123.0               Drama   nm8185151  
238746        2019            123.0               Drama   nm8185151  
238747        2019            123.0               Drama   nm8185151  
238748        2017            116.0                None  nm10538612  

[238749 rows x 7 columns]

In [49]:
merged_3= pd.merge(merged_temp,df_ratings, on=['movie_id'], how='inner')
merged_3

movie_id                    primary_title     original_title  \
0       tt0063540                        Sunghursh          Sunghursh   
1       tt0063540                        Sunghursh          Sunghursh   
2       tt0063540                        Sunghursh          Sunghursh   
3       tt0063540                        Sunghursh          Sunghursh   
4       tt0066787  One Day Before the Rainy Season    Ashad Ka Ek Din   
...           ...                              ...                ...   
163581  tt9913084                 Diabolik sono io   Diabolik sono io   
163582  tt9913084                 Diabolik sono io   Diabolik sono io   
163583  tt9914286                Sokagin Çocuklari  Sokagin Çocuklari   
163584  tt9914286                Sokagin Çocuklari  Sokagin Çocuklari   
163585  tt9916160                       Drømmeland         Drømmeland   

        start_year  runtime_minutes              genres  person_id  \
0             2013            175.0  Action,Crime,Drama  nm0712540   
1             2013            175.0  Action,Crime,Drama  nm0712540   
2             2013            175.0  Action,Crime,Drama  nm0712540   
3             2013            175.0  Action,Crime,Drama  nm0712540   
4             2019            114.0     Biography,Drama  nm0002411   
...            ...              ...                 ...        ...   
163581        2019             75.0         Documentary  nm0812850   
163582        2019             75.0         Documentary  nm0812850   
163583        2019             98.0        Drama,Family  nm4394529   
163584        2019             98.0        Drama,Family  nm4394529   
163585        2019             72.0         Documentary  nm5684093   

        averagerating  numvotes  
0                 7.0        77  
1                 7.0        77  
2                 7.0        77  
3                 7.0        77  
4                 7.2        43  
...               ...       ...  
163581            6.2         6  
163582            6.2         6  
163583            8.7       136  
163584            8.7       136  
163585            6.5        11  

[163586 rows x 9 columns]

In [50]:
merged_3.drop_duplicates(subset=['primary_title', 'original_title'])

movie_id                    primary_title  \
0       tt0063540                        Sunghursh   
4       tt0066787  One Day Before the Rainy Season   
5       tt0069049       The Other Side of the Wind   
7       tt0100275         The Wandering Soap Opera   
11      tt0137204                  Joe Finds Grace   
...           ...                              ...   
163575  tt9910930                      Jeg ser deg   
163577  tt9911774        Padmavyuhathile Abhimanyu   
163580  tt9913056                     Swarm Season   
163581  tt9913084                 Diabolik sono io   
163583  tt9914286                Sokagin Çocuklari   

                    original_title  start_year  runtime_minutes  \
0                        Sunghursh        2013            175.0   
4                  Ashad Ka Ek Din        2019            114.0   
5       The Other Side of the Wind        2018            122.0   
7            La Telenovela Errante        2017             80.0   
11                 Joe Finds Grace        2017             83.0   
...                            ...         ...              ...   
163575                 Jeg ser deg        2019             75.0   
163577   Padmavyuhathile Abhimanyu        2019            130.0   
163580                Swarm Season        2019             86.0   
163581            Diabolik sono io        2019             75.0   
163583           Sokagin Çocuklari        2019             98.0   

                            genres   person_id  averagerating  numvotes  
0               Action,Crime,Drama   nm0712540            7.0        77  
4                  Biography,Drama   nm0002411            7.2        43  
5                            Drama   nm0000080            6.9      4517  
7             Comedy,Drama,Fantasy   nm0765384            6.5       119  
11      Adventure,Animation,Comedy   nm0365480            8.1       263  
...                            ...         ...            ...       ...  
163575           Crime,Documentary  nm10536332            6.1         7  
163577                       Drama  nm10536451            8.4       365  
163580                 Documentary   nm1502645            6.2         5  
163581                 Documentary   nm0812850            6.2         6  
163583                Drama,Family   nm4394529            8.7       136  

[63652 rows x 9 columns]

In [51]:
merged_4= pd.merge(merged_temp,df_persons, on=['person_id'], how='inner')
merged_4

movie_id                    primary_title       original_title  \
0       tt0063540                        Sunghursh            Sunghursh   
1       tt0063540                        Sunghursh            Sunghursh   
2       tt0063540                        Sunghursh            Sunghursh   
3       tt0063540                        Sunghursh            Sunghursh   
4       tt0066787  One Day Before the Rainy Season      Ashad Ka Ek Din   
...           ...                              ...                  ...   
238742  tt9916190                        Safeguard            Safeguard   
238743  tt9916538              Kuambil Lagi Hatiku  Kuambil Lagi Hatiku   
238744  tt9916538              Kuambil Lagi Hatiku  Kuambil Lagi Hatiku   
238745  tt9916538              Kuambil Lagi Hatiku  Kuambil Lagi Hatiku   
238746  tt9916730                           6 Gunn               6 Gunn   

        start_year  runtime_minutes              genres   person_id  \
0             2013            175.0  Action,Crime,Drama   nm0712540   
1             2013            175.0  Action,Crime,Drama   nm0712540   
2             2013            175.0  Action,Crime,Drama   nm0712540   
3             2013            175.0  Action,Crime,Drama   nm0712540   
4             2019            114.0     Biography,Drama   nm0002411   
...            ...              ...                 ...         ...   
238742        2019             90.0      Drama,Thriller   nm7308376   
238743        2019            123.0               Drama   nm8185151   
238744        2019            123.0               Drama   nm8185151   
238745        2019            123.0               Drama   nm8185151   
238746        2017            116.0                None  nm10538612   

               primary_name        primary_profession  
0       Harnam Singh Rawail  director,writer,producer  
1       Harnam Singh Rawail  director,writer,producer  
2       Harnam Singh Rawail  director,writer,producer  
3       Harnam Singh Rawail  director,writer,producer  
4                 Mani Kaul     director,writer,actor  
...                     ...                       ...  
238742      Fraser Precious     actor,producer,writer  
238743    Azhar Kinoi Lubis                  director  
238744    Azhar Kinoi Lubis                  director  
238745    Azhar Kinoi Lubis                  director  
238746         Kiran Gawade                      None  

[238747 rows x 9 columns]

In [52]:
merged_4.drop_duplicates(subset=['person_id', 'primary_name'])

movie_id                                      primary_title  \
0       tt0063540                                          Sunghursh   
4       tt0066787                    One Day Before the Rainy Season   
5       tt0069049                         The Other Side of the Wind   
7       tt0100275                           The Wandering Soap Opera   
8       tt0100275                           The Wandering Soap Opera   
...           ...                                                ...   
238735  tt9915872                               The Last White Witch   
238738  tt9916170                                      The Rehearsal   
238741  tt9916186  Illenau - die Geschichte einer ehemaligen Heil...   
238742  tt9916190                                          Safeguard   
238746  tt9916730                                             6 Gunn   

                                           original_title  start_year  \
0                                               Sunghursh        2013   
4                                         Ashad Ka Ek Din        2019   
5                              The Other Side of the Wind        2018   
7                                   La Telenovela Errante        2017   
8                                   La Telenovela Errante        2017   
...                                                   ...         ...   
238735                               The Last White Witch        2019   
238738                                           O Ensaio        2019   
238741  Illenau - die Geschichte einer ehemaligen Heil...        2017   
238742                                          Safeguard        2019   
238746                                             6 Gunn        2017   

        runtime_minutes                genres   person_id  \
0                 175.0    Action,Crime,Drama   nm0712540   
4                 114.0       Biography,Drama   nm0002411   
5                 122.0                 Drama   nm0000080   
7                  80.0  Comedy,Drama,Fantasy   nm0765384   
8                  80.0  Comedy,Drama,Fantasy   nm0749914   
...                 ...                   ...         ...   
238735             97.0  Comedy,Drama,Fantasy   nm8063415   
238738             51.0                 Drama   nm5412267   
238741             84.0           Documentary  nm10538274   
238742             90.0        Drama,Thriller   nm7308376   
238746            116.0                  None  nm10538612   

               primary_name               primary_profession  
0       Harnam Singh Rawail         director,writer,producer  
4                 Mani Kaul            director,writer,actor  
5              Orson Welles            actor,director,writer  
7         Valeria Sarmiento           editor,director,writer  
8                Raoul Ruiz         director,writer,producer  
...                     ...                              ...  
238735        Hideki Kiyota           director,writer,editor  
238738      Tamar Guimaraes           director,writer,editor  
238741          Frank König  director,cinematographer,editor  
238742      Fraser Precious            actor,producer,writer  
238746         Kiran Gawade                             None  

[91331 rows x 9 columns]

In [53]:
df_basics.rename(columns = {'start_year':'release_date'}, inplace= True)
df_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   
5  tt0111414                      A Thin Life                 A Thin Life   

   release_date  runtime_minutes                genres  
0          2013            175.0    Action,Crime,Drama  
1          2019            114.0       Biography,Drama  
2          2018            122.0                 Drama  
4          2017             80.0  Comedy,Drama,Fantasy  
5          2018             75.0                Comedy

In [54]:
merged_2.rename(columns={'movie': 'primary_title'}, inplace=True)

In [55]:
merged_dataset = pd.merge(merged_2,df_basics, on =['primary_title','release_date'], how='inner')
merged_dataset

primary_title     studio  domestic_gross_x foreign_gross  \
0                    Toy Story 3         BV       415000000.0     652000000   
1                      Inception         WB       292600000.0     535700000   
2            Shrek Forever After       P/DW       238700000.0     513900000   
3     The Twilight Saga: Eclipse       Sum.       300500000.0     398000000   
4                     Iron Man 2       Par.       312400000.0     311500000   
...                          ...        ...               ...           ...   
1042                    Suspiria     Amazon         2500000.0       5400000   
1043         The Hurricane Heist      ENTMP         6100000.0             0   
1044                   Destroyer  Annapurna         1500000.0       4000000   
1045                       Gotti         VE         4300000.0             0   
1046                       Mandy        RLJ         1200000.0             0   

      release_date  id production_budget domestic_gross_y worldwide_gross  \
0             2010  47      $200,000,000     $415,004,880  $1,068,879,522   
1             2010  38      $160,000,000     $292,576,195    $835,524,642   
2             2010  27      $165,000,000     $238,736,787    $756,244,673   
3             2010  53       $68,000,000     $300,531,751    $706,102,828   
4             2010  15      $170,000,000     $312,433,331    $621,156,389   
...            ...  ..               ...              ...             ...   
1042          2018  81       $20,000,000       $2,483,472      $7,034,615   
1043          2018  22       $40,000,000       $6,115,824     $30,963,684   
1044          2018   5        $9,000,000       $1,533,324      $3,681,096   
1045          2018  64       $10,000,000       $4,286,367      $6,089,100   
1046          2018  71        $6,000,000       $1,214,525      $1,427,656   

       movie_id              original_title  runtime_minutes  \
0     tt0435761                 Toy Story 3            103.0   
1     tt1375666                   Inception            148.0   
2     tt0892791         Shrek Forever After             93.0   
3     tt1325004  The Twilight Saga: Eclipse            124.0   
4     tt1228705                  Iron Man 2            124.0   
...         ...                         ...              ...   
1042  tt1034415                    Suspiria            152.0   
1043  tt5360952         The Hurricane Heist            103.0   
1044  tt7137380                   Destroyer            121.0   
1045  tt1801552                       Gotti            112.0   
1046  tt6998518                       Mandy            121.0   

                          genres  
0     Adventure,Animation,Comedy  
1        Action,Adventure,Sci-Fi  
2     Adventure,Animation,Comedy  
3        Adventure,Drama,Fantasy  
4        Action,Adventure,Sci-Fi  
...                          ...  
1042      Fantasy,Horror,Mystery  
1043      Action,Adventure,Crime  
1044          Action,Crime,Drama  
1045       Biography,Crime,Drama  
1046       Action,Fantasy,Horror  

[1047 rows x 13 columns]

In [56]:
merged_dataset1=pd.merge(merged_dataset,df_ratings, on =['movie_id'], how='inner')
merged_dataset1

primary_title     studio  domestic_gross_x foreign_gross  \
0                    Toy Story 3         BV       415000000.0     652000000   
1                      Inception         WB       292600000.0     535700000   
2            Shrek Forever After       P/DW       238700000.0     513900000   
3     The Twilight Saga: Eclipse       Sum.       300500000.0     398000000   
4                     Iron Man 2       Par.       312400000.0     311500000   
...                          ...        ...               ...           ...   
1035                    Suspiria     Amazon         2500000.0       5400000   
1036         The Hurricane Heist      ENTMP         6100000.0             0   
1037                   Destroyer  Annapurna         1500000.0       4000000   
1038                       Gotti         VE         4300000.0             0   
1039                       Mandy        RLJ         1200000.0             0   

      release_date  id production_budget domestic_gross_y worldwide_gross  \
0             2010  47      $200,000,000     $415,004,880  $1,068,879,522   
1             2010  38      $160,000,000     $292,576,195    $835,524,642   
2             2010  27      $165,000,000     $238,736,787    $756,244,673   
3             2010  53       $68,000,000     $300,531,751    $706,102,828   
4             2010  15      $170,000,000     $312,433,331    $621,156,389   
...            ...  ..               ...              ...             ...   
1035          2018  81       $20,000,000       $2,483,472      $7,034,615   
1036          2018  22       $40,000,000       $6,115,824     $30,963,684   
1037          2018   5        $9,000,000       $1,533,324      $3,681,096   
1038          2018  64       $10,000,000       $4,286,367      $6,089,100   
1039          2018  71        $6,000,000       $1,214,525      $1,427,656   

       movie_id              original_title  runtime_minutes  \
0     tt0435761                 Toy Story 3            103.0   
1     tt1375666                   Inception            148.0   
2     tt0892791         Shrek Forever After             93.0   
3     tt1325004  The Twilight Saga: Eclipse            124.0   
4     tt1228705                  Iron Man 2            124.0   
...         ...                         ...              ...   
1035  tt1034415                    Suspiria            152.0   
1036  tt5360952         The Hurricane Heist            103.0   
1037  tt7137380                   Destroyer            121.0   
1038  tt1801552                       Gotti            112.0   
1039  tt6998518                       Mandy            121.0   

                          genres  averagerating  numvotes  
0     Adventure,Animation,Comedy            8.3    682218  
1        Action,Adventure,Sci-Fi            8.8   1841066  
2     Adventure,Animation,Comedy            6.3    167532  
3        Adventure,Drama,Fantasy            5.0    211733  
4        Action,Adventure,Sci-Fi            7.0    657690  
...                          ...            ...       ...  
1035      Fantasy,Horror,Mystery            6.8     37360  
1036      Action,Adventure,Crime            5.1     15326  
1037          Action,Crime,Drama            6.2     13683  
1038       Biography,Crime,Drama            4.8     10358  
1039       Action,Fantasy,Horror            6.6     44378  

[1040 rows x 15 columns]

In [57]:
merged_dataset2=pd.merge(merged_dataset1,df_directors, on =['movie_id'], how='left')
merged_dataset2

primary_title studio  domestic_gross_x foreign_gross  release_date  id  \
0      Toy Story 3     BV       415000000.0     652000000          2010  47   
1      Toy Story 3     BV       415000000.0     652000000          2010  47   
2      Toy Story 3     BV       415000000.0     652000000          2010  47   
3      Toy Story 3     BV       415000000.0     652000000          2010  47   
4        Inception     WB       292600000.0     535700000          2010  38   
...            ...    ...               ...           ...           ...  ..   
3431         Gotti     VE         4300000.0             0          2018  64   
3432         Gotti     VE         4300000.0             0          2018  64   
3433         Mandy    RLJ         1200000.0             0          2018  71   
3434         Mandy    RLJ         1200000.0             0          2018  71   
3435         Mandy    RLJ         1200000.0             0          2018  71   

     production_budget domestic_gross_y worldwide_gross   movie_id  \
0         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
1         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
2         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
3         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
4         $160,000,000     $292,576,195    $835,524,642  tt1375666   
...                ...              ...             ...        ...   
3431       $10,000,000       $4,286,367      $6,089,100  tt1801552   
3432       $10,000,000       $4,286,367      $6,089,100  tt1801552   
3433        $6,000,000       $1,214,525      $1,427,656  tt6998518   
3434        $6,000,000       $1,214,525      $1,427,656  tt6998518   
3435        $6,000,000       $1,214,525      $1,427,656  tt6998518   

     original_title  runtime_minutes                      genres  \
0       Toy Story 3            103.0  Adventure,Animation,Comedy   
1       Toy Story 3            103.0  Adventure,Animation,Comedy   
2       Toy Story 3            103.0  Adventure,Animation,Comedy   
3       Toy Story 3            103.0  Adventure,Animation,Comedy   
4         Inception            148.0     Action,Adventure,Sci-Fi   
...             ...              ...                         ...   
3431          Gotti            112.0       Biography,Crime,Drama   
3432          Gotti            112.0       Biography,Crime,Drama   
3433          Mandy            121.0       Action,Fantasy,Horror   
3434          Mandy            121.0       Action,Fantasy,Horror   
3435          Mandy            121.0       Action,Fantasy,Horror   

      averagerating  numvotes  person_id  
0               8.3    682218  nm0881279  
1               8.3    682218  nm0881279  
2               8.3    682218  nm0881279  
3               8.3    682218  nm0881279  
4               8.8   1841066  nm0634240  
...             ...       ...        ...  
3431            4.8     10358  nm0175305  
3432            4.8     10358  nm0175305  
3433            6.6     44378  nm0181903  
3434            6.6     44378  nm0181903  
3435            6.6     44378  nm0181903  

[3436 rows x 16 columns]

In [66]:
merged_dataset3=pd.merge(merged_dataset2,df_persons, on =['person_id'], how='left')
merged_dataset3

primary_title studio  domestic_gross_x foreign_gross  release_date  id  \
0      Toy Story 3     BV       415000000.0     652000000          2010  47   
1      Toy Story 3     BV       415000000.0     652000000          2010  47   
2      Toy Story 3     BV       415000000.0     652000000          2010  47   
3      Toy Story 3     BV       415000000.0     652000000          2010  47   
4        Inception     WB       292600000.0     535700000          2010  38   
...            ...    ...               ...           ...           ...  ..   
3431         Gotti     VE         4300000.0             0          2018  64   
3432         Gotti     VE         4300000.0             0          2018  64   
3433         Mandy    RLJ         1200000.0             0          2018  71   
3434         Mandy    RLJ         1200000.0             0          2018  71   
3435         Mandy    RLJ         1200000.0             0          2018  71   

     production_budget domestic_gross_y worldwide_gross   movie_id  \
0         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
1         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
2         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
3         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
4         $160,000,000     $292,576,195    $835,524,642  tt1375666   
...                ...              ...             ...        ...   
3431       $10,000,000       $4,286,367      $6,089,100  tt1801552   
3432       $10,000,000       $4,286,367      $6,089,100  tt1801552   
3433        $6,000,000       $1,214,525      $1,427,656  tt6998518   
3434        $6,000,000       $1,214,525      $1,427,656  tt6998518   
3435        $6,000,000       $1,214,525      $1,427,656  tt6998518   

     original_title  runtime_minutes                      genres  \
0       Toy Story 3            103.0  Adventure,Animation,Comedy   
1       Toy Story 3            103.0  Adventure,Animation,Comedy   
2       Toy Story 3            103.0  Adventure,Animation,Comedy   
3       Toy Story 3            103.0  Adventure,Animation,Comedy   
4         Inception            148.0     Action,Adventure,Sci-Fi   
...             ...              ...                         ...   
3431          Gotti            112.0       Biography,Crime,Drama   
3432          Gotti            112.0       Biography,Crime,Drama   
3433          Mandy            121.0       Action,Fantasy,Horror   
3434          Mandy            121.0       Action,Fantasy,Horror   
3435          Mandy            121.0       Action,Fantasy,Horror   

      averagerating  numvotes  person_id       primary_name  \
0               8.3    682218  nm0881279        Lee Unkrich   
1               8.3    682218  nm0881279        Lee Unkrich   
2               8.3    682218  nm0881279        Lee Unkrich   
3               8.3    682218  nm0881279        Lee Unkrich   
4               8.8   1841066  nm0634240  Christopher Nolan   
...             ...       ...        ...                ...   
3431            4.8     10358  nm0175305     Kevin Connolly   
3432            4.8     10358  nm0175305     Kevin Connolly   
3433            6.6     44378  nm0181903     Panos Cosmatos   
3434            6.6     44378  nm0181903     Panos Cosmatos   
3435            6.6     44378  nm0181903     Panos Cosmatos   

                             primary_profession  
0     editorial_department,editor,miscellaneous  
1     editorial_department,editor,miscellaneous  
2     editorial_department,editor,miscellaneous  
3     editorial_department,editor,miscellaneous  
4                      writer,producer,director  
...                                         ...  
3431                    actor,director,producer  
3432                    actor,director,producer  
3433                   writer,director,producer  
3434                   writer,director,producer  
3435                   writer,director,producer  

[3436 rows x 18 columns]

In [67]:
final= merged_dataset3.drop(columns=['original_title'], errors= 'ignore')
final


primary_title studio  domestic_gross_x foreign_gross  release_date  id  \
0      Toy Story 3     BV       415000000.0     652000000          2010  47   
1      Toy Story 3     BV       415000000.0     652000000          2010  47   
2      Toy Story 3     BV       415000000.0     652000000          2010  47   
3      Toy Story 3     BV       415000000.0     652000000          2010  47   
4        Inception     WB       292600000.0     535700000          2010  38   
...            ...    ...               ...           ...           ...  ..   
3431         Gotti     VE         4300000.0             0          2018  64   
3432         Gotti     VE         4300000.0             0          2018  64   
3433         Mandy    RLJ         1200000.0             0          2018  71   
3434         Mandy    RLJ         1200000.0             0          2018  71   
3435         Mandy    RLJ         1200000.0             0          2018  71   

     production_budget domestic_gross_y worldwide_gross   movie_id  \
0         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
1         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
2         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
3         $200,000,000     $415,004,880  $1,068,879,522  tt0435761   
4         $160,000,000     $292,576,195    $835,524,642  tt1375666   
...                ...              ...             ...        ...   
3431       $10,000,000       $4,286,367      $6,089,100  tt1801552   
3432       $10,000,000       $4,286,367      $6,089,100  tt1801552   
3433        $6,000,000       $1,214,525      $1,427,656  tt6998518   
3434        $6,000,000       $1,214,525      $1,427,656  tt6998518   
3435        $6,000,000       $1,214,525      $1,427,656  tt6998518   

      runtime_minutes                      genres  averagerating  numvotes  \
0               103.0  Adventure,Animation,Comedy            8.3    682218   
1               103.0  Adventure,Animation,Comedy            8.3    682218   
2               103.0  Adventure,Animation,Comedy            8.3    682218   
3               103.0  Adventure,Animation,Comedy            8.3    682218   
4               148.0     Action,Adventure,Sci-Fi            8.8   1841066   
...               ...                         ...            ...       ...   
3431            112.0       Biography,Crime,Drama            4.8     10358   
3432            112.0       Biography,Crime,Drama            4.8     10358   
3433            121.0       Action,Fantasy,Horror            6.6     44378   
3434            121.0       Action,Fantasy,Horror            6.6     44378   
3435            121.0       Action,Fantasy,Horror            6.6     44378   

      person_id       primary_name                         primary_profession  
0     nm0881279        Lee Unkrich  editorial_department,editor,miscellaneous  
1     nm0881279        Lee Unkrich  editorial_department,editor,miscellaneous  
2     nm0881279        Lee Unkrich  editorial_department,editor,miscellaneous  
3     nm0881279        Lee Unkrich  editorial_department,editor,miscellaneous  
4     nm0634240  Christopher Nolan                   writer,producer,director  
...         ...                ...                                        ...  
3431  nm0175305     Kevin Connolly                    actor,director,producer  
3432  nm0175305     Kevin Connolly                    actor,director,producer  
3433  nm0181903     Panos Cosmatos                   writer,director,producer  
3434  nm0181903     Panos Cosmatos                   writer,director,producer  
3435  nm0181903     Panos Cosmatos                   writer,director,producer  

[3436 rows x 17 columns]

In [71]:
final=final.drop_duplicates(subset=['primary_title', 'release_date'])
final

primary_title     studio  domestic_gross_x foreign_gross  \
0                    Toy Story 3         BV       415000000.0     652000000   
4                      Inception         WB       292600000.0     535700000   
5            Shrek Forever After       P/DW       238700000.0     513900000   
15    The Twilight Saga: Eclipse       Sum.       300500000.0     398000000   
17                    Iron Man 2       Par.       312400000.0     311500000   
...                          ...        ...               ...           ...   
3422                    Suspiria     Amazon         2500000.0       5400000   
3425         The Hurricane Heist      ENTMP         6100000.0             0   
3429                   Destroyer  Annapurna         1500000.0       4000000   
3431                       Gotti         VE         4300000.0             0   
3433                       Mandy        RLJ         1200000.0             0   

      release_date  id production_budget domestic_gross_y worldwide_gross  \
0             2010  47      $200,000,000     $415,004,880  $1,068,879,522   
4             2010  38      $160,000,000     $292,576,195    $835,524,642   
5             2010  27      $165,000,000     $238,736,787    $756,244,673   
15            2010  53       $68,000,000     $300,531,751    $706,102,828   
17            2010  15      $170,000,000     $312,433,331    $621,156,389   
...            ...  ..               ...              ...             ...   
3422          2018  81       $20,000,000       $2,483,472      $7,034,615   
3425          2018  22       $40,000,000       $6,115,824     $30,963,684   
3429          2018   5        $9,000,000       $1,533,324      $3,681,096   
3431          2018  64       $10,000,000       $4,286,367      $6,089,100   
3433          2018  71        $6,000,000       $1,214,525      $1,427,656   

       movie_id  runtime_minutes                      genres  averagerating  \
0     tt0435761            103.0  Adventure,Animation,Comedy            8.3   
4     tt1375666            148.0     Action,Adventure,Sci-Fi            8.8   
5     tt0892791             93.0  Adventure,Animation,Comedy            6.3   
15    tt1325004            124.0     Adventure,Drama,Fantasy            5.0   
17    tt1228705            124.0     Action,Adventure,Sci-Fi            7.0   
...         ...              ...                         ...            ...   
3422  tt1034415            152.0      Fantasy,Horror,Mystery            6.8   
3425  tt5360952            103.0      Action,Adventure,Crime            5.1   
3429  tt7137380            121.0          Action,Crime,Drama            6.2   
3431  tt1801552            112.0       Biography,Crime,Drama            4.8   
3433  tt6998518            121.0       Action,Fantasy,Horror            6.6   

      numvotes  person_id       primary_name  \
0       682218  nm0881279        Lee Unkrich   
4      1841066  nm0634240  Christopher Nolan   
5       167532  nm0593610      Mike Mitchell   
15      211733  nm1720541        David Slade   
17      657690  nm0269463        Jon Favreau   
...        ...        ...                ...   
3422     37360  nm0345174    Luca Guadagnino   
3425     15326  nm0003418          Rob Cohen   
3429     13683  nm0476201       Karyn Kusama   
3431     10358  nm0175305     Kevin Connolly   
3433     44378  nm0181903     Panos Cosmatos   

                             primary_profession  
0     editorial_department,editor,miscellaneous  
4                      writer,producer,director  
5                         actor,director,writer  
15                     producer,director,writer  
17                 producer,actor,miscellaneous  
...                                         ...  
3422                   director,producer,writer  
3425                    producer,director,actor  
3429                   director,writer,producer  
3431                    actor,director,producer  
3433                   writer,director,producer  

[1025 rows x 17 co

# EDA